File: readability/readabilty.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string passage = get_string("Text: ");
    float sentence_counter = 0;
    float letter_counter = 0;
    float word_counter = 1;
    float average_number_letters = 0;
    float average_number_sentence = 0;
    float reading_score = 0;

     for(int index = 0; index < strlen(passage); index++)
    {
        if(passage[index] == ' ')
        {
            word_counter++;
        }
        else if(passage[index] == '.' || passage[index] == '!' || passage[index] == '?')
        {
            sentence_counter++;
        }
        else
        {
              if(passage[index] >=65 && passage[index] <=122)
              {
                letter_counter++;
              }
        }
    }
    average_number_letters = ((float)letter_counter / word_counter) * 100;
    average_number_sentence = ((float)sentence_counter / word_counter) *100;
    reading_score = (0.0588 * average_number_letters) - (0.296 * average_number_sentence) - 15.8;

    if(reading_score < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (reading_score > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)round(reading_score));
    }

    //printf("let %f  word %f  sen %f\n", letter_counter, word_counter, sentence_counter);
}

File: Week_6/sentimental-readability/readabilty.py
from cs50 import get_string
import math

letter = 0
word = 1
sentence = 0
text = get_string("Text: ")

for x in text:
    if x == " ":
        word += 1
    elif x == "." or x == "!" or x == "?":
        sentence += 1
    else:
        letter += 1

cole_L = letter / word * 100
cole_S = sentence / word * 100
cole_answer = 0.0588 * cole_L - 0.296 * cole_S - 15.8

if cole_answer > 16:
    print("Grade 16+")
elif 1 <= cole_answer <= 16:
    print(f"Grade {math.floor(cole_answer+.03)}")
else:
    print("Before Grade 1")
